### DISCUSSION TITLE
Allow specifying -C twice/thrice when blaming

### DISCUSSION THREAD

**Discussion by felipeochoa** [magit/magit#5061] [Ideas]
git-blame [allows](https://www.git-scm.com/docs/git-blame) specifying `-C` up to three times:

> When this option is given twice, the command additionally looks for copies from other files in the commit that creates the file.
> 
> When this option is given three times, the command additionally looks for copies from other files in any commit.

I don't think there's a way to pass this flag more than once via transient, so I've resorted to doing this:

```elisp
(transient-replace-suffix
  'magit-blame
  '(0 5)
  '("-D" "Look for copies from other files" "-C"))
```

Which allows using `-D` to add another `-C` flag. It works for me, but it's a bit hacky and it would be nice if this was built-into magit. Some quirks to deal with would be:

* Only one of the `-C` flags should take the (optional) number argument "If there are more than one `-C` options given, the `<num>` argument of the last `-C` will take effect."
* The transient menu would ideally allow cycling between 0, 1, 2, or 3 uses of the `-C` flag.

**Comment by tarsius** [comment]
Those are funky arguments, aren't they?!

I'll have to think about how to best represent the available behaviors in Magit. I think a 1to1 one mapping won't do in this case, so this will probably involve a new specialized Transient suffix class.

I'm trying to wrap things up this month, so I am very unlikely to work on it this month.

**Reply by felipeochoa** [reply]
> Those are funky arguments, aren't they?!

You said it!

I have a very limited understanding of transient, but I think the most sensible way to deal with these would be to use prefix arguments, so that:

* `C-1 - C` or just `- C` prompts for a numeric argument and enables `-C`  (current behavior)
* `C-2 - C` prompts for a number and enables `-C -C`
* `C-3 - C` prompts for a number and enalbes `-C -C -C`

Another idea would be to have `-C` cycle through `-C`, `-C -C`, `-C -C -C` and finally nothing.
